### PR TITLE
Two-stage GS support for multi-component vectors

### PR DESCRIPTION
**Pull request by victorapm**
Updated from #736. This PR enables 2stage GS relaxation to work properly with multi-component vectors.

Regression tests:

- [x] Lassen
- [x] Tux
- [x] Jlse

### CODE REVIEW

**Comment by PaulMullowney** [comment]
Any update on getting this merged?

**Comment by victorapm** [comment]
Hi @PaulMullowney, I'm doing some performance analysis with the new kernels added here. I'll need to change the strategy for loop unrolling and add regression tests. I expect to finish the work and mark this PR as ready for review by the end of this week. Hope this is okay for you. Thanks!

**Comment by PaulMullowney** [comment]
Are you talking about DiagScaleVector2?

**Comment by victorapm** [comment]
Right, and `hypre_DiagScaleVector` as well. I'm taking care of that

**Comment by PaulMullowney** [comment]
I guess the write to z with += repeatedly is kinda silly. ONe should save in registers and then write in 1 operation.

**Comment by PaulMullowney** [comment]
Sorry to bother again. Any sense of when this will get merged? I want to start moving these features into actual usage in ExaWind.

**Comment by victorapm** [comment]
@PaulMullowney sorry for the delay, I'm working on this right now. I want to mark this PR as ready for review today.

File: src/utilities/device_utils.c [hunk 1]
```diff
@@ -2188,7 +2187,7 @@ hypre_CudaCompileFlagCheck()
 /*--------------------------------------------------------------------
  * hypreGPUKernel_DiagScaleVector
  *
- * Assumes vectors components are stored contiguously.
+ * Assumes vectors components are stored columwise.
```
Comment (rfalgout) [inline src/utilities/device_utils.c]: Neither of these comments makes sense to me.  Is it even necessary?  This is a multivector, and the way multivector data is stored only needs to be documented once in an appropriate header file.

**Review by rfalgout — APPROVED** [review]
Looks good to me.  Thanks!

File: src/utilities/device_utils.c (hunk 1 above)
Comment (victorapm) [reply src/utilities/device_utils.c]: Yes, agreed!

**Comment by waynemitchell** [comment]
No issues with this PR for the sycl build on JLSE.

**Comment by PaulMullowney** [comment]
Thanks for getting this in!

**Pull request merged**